### DISCUSSION TITLE
[Feature] Map positions

### DISCUSSION THREAD

**Discussion by ghost** [immich-app/immich#4705] [Feature Request]
### The feature

when the map page opens its not even showing my country in the view. Every time it's opened the map has to be moved around.

any way to have a setting for a centre starting point, and zoom size for when the map is opened?

and while the app is running save a current position and zoom, so when clicking back to the map page it is not reset and users have to start moving and zooming again

Australia is not even in view
<img width="1659" alt="Screenshot 2023-10-30 at 10 08 25 am" src="https://github.com/immich-app/immich/assets/1471841/d149b310-00f9-4481-a2fc-8f55c55dd08b">

cheers

### Platform

- [ ] Server
- [x] Web
- [x] Mobile

**Comment by radh21301** [comment]
I think this would be really helpful. Especially in the mobile app. Currently it starts up with a  huge portion of the world map, and hence querying/loading almost all my assets (about 40k). This also takes a couple of seconds. Having a region (or the current location) to select as the default position would be a better way to optimize this I think.

**Comment by lefuturiste** [comment]
What if, for a starter, we can have a setting to automatically zoom to the bounding boxes of the pictures we want to show on the map? That will allow user to have the best map without setting a default position
Then we can have another setting that set a default position. And even better would be to have different locations presets.
I will be using this feature a lot as an OpenStreetMap contributor that take a lot of pictures.

**Reply by lefuturiste** [reply]
I will be happy to contribute on this specific feature.

**Reply by ghost** [reply]
Default point and zoom would be a great start 😃

**Reply by lefuturiste** [reply]
well, will be working on a PR soon for that

**Reply by martabal** [reply]
Maybe wait for https://github.com/immich-app/immich/pull/4294 to be merged before working on it

**Reply by lefuturiste** [reply]
yes good idea, there migrating from leaflet to maplibre so this will change the way we do that.

**Reply by lefuturiste** [reply]
Now I can start working on it :)

**Comment by msdevega** [comment]
I believe that add initial position and zoom of the map linked to an album it's a must. Any advance in this feature?
I can help in testing if it is needed with different OS (Linux, OSX).
